lumigator/backend/backend/repositories/secrets.py

```diff
@@ -27,6 +28,31 @@ def get_secret_by_name(self, name: str) -> SecretRecord | None:
     def list_secrets(self) -> list[SecretRecord]:
         """List all secrets."""
         return self.list()
+    
+    
+    def _create_secret(self, secret_data: dict) -> SecretRecord:
+        """Create a new secret."""
+        secret = SecretRecord(**secret_data)
+        self.session.add(secret)
+        self.session.commit()
+        loguru.logger.critical(f"DB _create_secret --> {[(r.name, r.value, r.description) for r in self.list()]}")
+
+        return secret
+
+    def _update_secret(self, secret_id: UUID, secret_data: dict) -> SecretRecord | None:
+        """Update an existing secret."""
+        secret = self.session.query(SecretRecord).filter(SecretRecord.id == secret_id).first()
+        if not secret:
+            return None
+
+        for key, value in secret_data.items():
+            setattr(secret, key, value)
+        self.session.commit()
+        loguru.logger.critical(
+            f"DB _update_secret --> {[(r.name, r._name, r.value, r.description) for r in self.list()]}"
+        )
+
+        return secret
```

Comment:
```suggestion
```

This is duplicated from below (without the logging of the secret)

Reply:
Ok, removing duplicate sections (merge issues I presume)